pnpm fails to install Node.js on 64 bit Raspbian

<!--
  If this issue affects many people in a company/big team, create a post for your company in the following discussion:
  https://github.com/pnpm/pnpm/discussions/3787
  and link the issue in your post.

  This will help us prioritize issues that affect more people.
-->

### pnpm version: 6.22.2

### Code to reproduce the issue:
```
pnpm env use -g latest
```
<!--
If there was a fatal error also include a gist of your node_modules/.pnpm-debug.log file.
-->

### Expected behavior:
pnpm downloads and installs the latest version of node

### Actual behavior:
pnpm outputs a error message (ERR_PNPM_FETCH_404) and fails:
```
 WARN  GET https://nodejs.org/download/release/v17.1.0/node-v17.1.0-linux-arm.tar.gz error (ERR_PNPM_FETCH_404). Will retry in 10 seconds. 2 retries left.
 WARN  GET https://nodejs.org/download/release/v17.1.0/node-v17.1.0-linux-arm.tar.gz error (ERR_PNPM_FETCH_404). Will retry in 1 minute. 1 retries left.
 ERR_PNPM_FETCH_404  GET https://nodejs.org/download/release/v17.1.0/node-v17.1.0-linux-arm.tar.gz: Not Found - 404
```

### Additional information:

 - `node -v` prints: `v16.10.0`
 - Windows, macOS, or Linux?: Linux (64 bit Rasbian OS on a Raspberry Pi 4)

--- a/.changeset/odd-garlics-laugh.md
+++ b/.changeset/odd-garlics-laugh.md
@@ -1,0 +1,6 @@
+---
+"@pnpm/plugin-commands-env": patch
+"pnpm": patch
+---
+
+`pnpm env use` should download the right Node.js tarball on Raspberry Pi [#4007](https://github.com/pnpm/pnpm/issues/4007).

--- a/packages/plugin-commands-env/src/node.ts
+++ b/packages/plugin-commands-env/src/node.ts
@@ -11,6 +11,7 @@ import renameOverwrite from 'rename-overwrite'
 import tempy from 'tempy'
 import loadJsonFile from 'load-json-file'
 import writeJsonFile from 'write-json-file'
+import normalizeArch from './normalizeArch'
 
 export type NvmNodeCommandOptions = Pick<Config,
 | 'bin'
@@ -116,23 +117,13 @@ async function downloadAndUnpackZip (
 
 function getNodeJSTarball (nodeVersion: string, releaseDir: string) {
   const platform = process.platform === 'win32' ? 'win' : process.platform
-  const arch = getArch(platform)
+  const arch = normalizeArch(process.platform, process.arch)
   const extension = platform === 'win' ? 'zip' : 'tar.gz'
   const pkgName = `node-v${nodeVersion}-${platform}-${arch}`
   return {
     pkgName,
     tarball: `https://nodejs.org/download/${releaseDir}/v${nodeVersion}/${pkgName}.${extension}`,
   }
-}
-
-function getArch (platform: string) {
-  if (platform === 'win' && process.arch === 'ia32') {
-    return 'x86'
-  }
-  if (process.arch === 'arm') {
-    return 'armv7l'
-  }
-  return process.arch
 }
 
 async function readNodeVersionsManifest (nodesDir: string): Promise<{ default?: string }> {

--- a/packages/plugin-commands-env/src/node.ts
+++ b/packages/plugin-commands-env/src/node.ts
@@ -116,13 +116,23 @@ async function downloadAndUnpackZip (
 
 function getNodeJSTarball (nodeVersion: string, releaseDir: string) {
   const platform = process.platform === 'win32' ? 'win' : process.platform
-  const arch = platform === 'win' && process.arch === 'ia32' ? 'x86' : process.arch
+  const arch = getArch(platform)
   const extension = platform === 'win' ? 'zip' : 'tar.gz'
   const pkgName = `node-v${nodeVersion}-${platform}-${arch}`
   return {
     pkgName,
     tarball: `https://nodejs.org/download/${releaseDir}/v${nodeVersion}/${pkgName}.${extension}`,
   }
+}
+
+function getArch (platform: string) {
+  if (platform === 'win' && process.arch === 'ia32') {
+    return 'x86'
+  }
+  if (process.arch === 'arm') {
+    return 'armv7l'
+  }
+  return process.arch
 }
 
 async function readNodeVersionsManifest (nodesDir: string): Promise<{ default?: string }> {

--- a/packages/plugin-commands-env/src/normalizeArch.ts
+++ b/packages/plugin-commands-env/src/normalizeArch.ts
@@ -1,0 +1,9 @@
+export default function getNormalizedArch (platform: string, arch: string) {
+  if (platform === 'win32' && arch === 'ia32') {
+    return 'x86'
+  }
+  if (arch === 'arm') {
+    return 'armv7l'
+  }
+  return arch
+}

--- a/packages/plugin-commands-env/test/normalizeArch.test.ts
+++ b/packages/plugin-commands-env/test/normalizeArch.test.ts
@@ -1,0 +1,9 @@
+import normalizeArch from '@pnpm/plugin-commands-env/lib/normalizeArch'
+
+test.each([
+  ['win32', 'ia32', 'x86'],
+  ['linux', 'arm', 'armv7l'], // Raspberry Pi 4
+  ['linux', 'x64', 'x64'],
+])('%s is normalized to %s', (platform, arch, normalizedArch) => {
+  expect(normalizeArch(platform, arch)).toBe(normalizedArch)
+})

--- a/packages/plugin-commands-env/test/normalizeArch.test.ts
+++ b/packages/plugin-commands-env/test/normalizeArch.test.ts
@@ -4,6 +4,6 @@ test.each([
   ['win32', 'ia32', 'x86'],
   ['linux', 'arm', 'armv7l'], // Raspberry Pi 4
   ['linux', 'x64', 'x64'],
-])('%s is normalized to %s', (platform, arch, normalizedArch) => {
+])('normalizedArch(%s, %s)', (platform, arch, normalizedArch) => {
   expect(normalizeArch(platform, arch)).toBe(normalizedArch)
 })

